httpx_cache/cache/redis.py

```diff
@@ -0,0 +1,102 @@
+import threading
+import typing as tp
+from abc import ABC
+
+import anyio
+import httpx
+from httpx_cache.cache.base import BaseCache
+from httpx_cache.serializer.base import BaseSerializer
+from httpx_cache.serializer.common import MsgPackSerializer
+from httpx_cache.utils import get_cache_key
+from redis import Redis
+from redis.asyncio import Redis as AsyncRedis
```

Comment:
Missing the `redis` dependency in `pyproject.toml`, I can add it as an optional dep in a separate PR